src/reference/04-Howto/20-Howto-Sequencing/04-Howto-Dynamic-Input-Task.md

```diff
@@ -0,0 +1,61 @@
+---
+out: Howto-Dynamic-Input-Task.html
+---
+
+### Defining a dynamic input task with Def.inputTaskDyn
+
+Let's suppose that there's a task already that does the bowser opening called `openbrowser` because of a plugin. Here's how we can sequence a task after an input tasks.
+
+#### build.sbt
+
+```scala
+lazy val runopen = inputKey[Unit]("run and then open the browser")
+lazy val openbrowser = taskKey[Unit]("open the browser")
+
+lazy val root = (project in file(".")).
+  settings(
+    runopen := (Def.inputTaskDyn {
+      import sbt.complete.Parsers.spaceDelimited
+      val args = spaceDelimited("<args>").parsed
+      Def.taskDyn {
+        (run in Compile).toTask(" " + args.mkString(" ")).value
+        openbrowser
+      }
+    }).evaluated,
+    openbrowser := {
+      println("open browser!")
+    }
+  )
+```
+
+#### build.sbt v2
+
+Trying to rewire `run in Compile` is going to be complicated. Since the reference to the inner `run in Compile` is already inside the continuation task, simply rewiring `runopen` to `run in Compile` will create a cyclic reference.
+To break the cycle, we will introduce a clone of `run in Compile` called `actualRun in Compile`:
+
+```scala
+lazy val actualRun = inputKey[Unit]("The actual run task")
+lazy val openbrowser = taskKey[Unit]("open the browser")
+
+lazy val root = (project in file(".")).
+  settings(
+    run in Compile := (Def.inputTaskDyn {
+      import sbt.complete.Parsers.spaceDelimited
+      val args = spaceDelimited("<args>").parsed
+      Def.taskDyn {
+        (actualRun in Compile).toTask(" " + args.mkString(" ")).value
+        openbrowser
+      }
+    }).evaluated,
+    actualRun in Compile := Defaults.runTask(
```

Comment:
It'd be helpful if this tutorial explained how you came up with the definition of `actualRun in Compile`.